### PR TITLE
Embeded independent module in repo for datasource

### PR DESCRIPTION
**Pull request by louyuting**
### Special notes for reviews

### CODE REVIEW

**Comment by louyuting** [comment]
@sczyh30  Could you please take a look?

**Comment by sczyh30** [comment]
> @sczyh30 Could you please take a look?

I'll verify it this weekend.

**Review by sanxun0325 — APPROVED** [review]
LGTM

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -55,6 +55,14 @@ jobs:
         diff -u <(echo -n) <(gofmt -d -s .)
         diff -u <(echo -n) <(goimports -d .)
         go test -v -race ./... -coverprofile=coverage.txt -covermode=atomic
+        cd ./pkg/datasource/consul
```
Comment (sczyh30) [inline .github/workflows/go.yml:58]: Maybe we could improve here later.

**Review by sczyh30 — APPROVED** [review]

**Comment by sczyh30** [comment]
This could bring breaking change (because the pkg path has been changed). We might need to find a good way to tackle the compatibility problem later.

File: .github/workflows/go.yml (hunk 1 above)
Comment (louyuting) [reply .github/workflows/go.yml:58]: echo

**Comment by louyuting** [comment]
> This could bring breaking change (because the pkg path has been changed). We might need to find a good way to tackle the compatibility problem later.

Maybe we could keep release version compatibility. 
These datasource and adapter need release synchronously

**Pull request merged**